File: apps/desktop/src/services/NativeService/appUse.ts
import { invoke } from '@tauri-apps/api/core';

import type {
    AppUseNativeActRequest,
    AppUseNativeActResponse,
    AppUseNativeAuthorizeActRequest,
    AppUseNativeAuthorizeActResponse,
    AppUseNativeObserveRequest,
    AppUseNativeObserveResponse,
    AppUseNativeSessionRequest,
    AppUseNativeSessionResponse,
} from './types';

export const appUse = {
    session(request: AppUseNativeSessionRequest): Promise<AppUseNativeSessionResponse> {
        return invoke('app_use_session', { request });
    },
    observe(request: AppUseNativeObserveRequest): Promise<AppUseNativeObserveResponse> {
        return invoke('app_use_observe', { request });
    },
    authorizeAct(
        request: AppUseNativeAuthorizeActRequest
    ): Promise<AppUseNativeAuthorizeActResponse> {
        return invoke('app_use_authorize_act', { request });
    },
    act(request: AppUseNativeActRequest): Promise<AppUseNativeActResponse> {
        return invoke('app_use_act', { request });
    },
} as const;

File: apps/desktop/src-tauri/src/commands/app_use.rs
// Copyright (c) 2026. Qian Cheng. Licensed under GPL v3

use crate::core::app_use::{
    AppUseActRequest, AppUseActResponse, AppUseAuthorizeActRequest, AppUseAuthorizeActResponse,
    AppUseObserveRequest, AppUseObserveResponse, AppUseRuntime, AppUseSessionRequest,
    AppUseSessionResponse,
};
use tauri::State;

#[tauri::command]
pub fn app_use_session(
    request: AppUseSessionRequest,
    runtime: State<'_, AppUseRuntime>,
) -> Result<AppUseSessionResponse, String> {
    Ok(runtime.session(request))
}

#[tauri::command]
pub fn app_use_observe(
    request: AppUseObserveRequest,
    runtime: State<'_, AppUseRuntime>,
) -> Result<AppUseObserveResponse, String> {
    Ok(runtime.observe(request))
}

#[tauri::command]
pub fn app_use_authorize_act(
    request: AppUseAuthorizeActRequest,
    runtime: State<'_, AppUseRuntime>,
) -> Result<AppUseAuthorizeActResponse, String> {
    Ok(runtime.authorize_act(request))
}

#[tauri::command]
pub fn app_use_act(
    request: AppUseActRequest,
    runtime: State<'_, AppUseRuntime>,
) -> Result<AppUseActResponse, String> {
    Ok(runtime.act(request))
}
